[amazonechocontrol] Add do not disturb channel

ccutrer opened the pull request:
This is a port of https://github.com/smarthomej/addons/pull/391 that I've been using for years, but I hear the main distro version of the binding should be functional again, and I'd rather not have to wait for it to become compatible after a new core release each time again.

J-N-K commented:
I would prefer if you backport all changes. I just don't have time for that at the Moment. I can help if Problems occur.

ccutrer commented:
Sure, I can help you with that. I still need to test my other use cases besides DnD. I'll work on the rest of them over the next two weeks. Do you prefer if I stuff it all into a single PR, or do PR per "topic" that happened in smarthomej?

Tangentially- what are your plans for the Tuya binding? I'd be happy to work on porting that into the main distro as well.

J-N-K commented:
@ccutrer I think that it would be better to do single PR because some of the PR had issues that were later fixed together in another PR. What I did when backporting other changes is to just replace the openhab version with the smarthomej version, rename the directory, do a search&replace for `org.smarthomej` and then check fi something unwanted was reverted.

J-N-K commented:
Regarding Tuya: fine with me.

ccutrer commented:
@J-N-K: okay, full binding ported in. Honestly, the hardest conflict was the README, since both sides had had significant work there. 🤞 I kept the essence of updates on both sides. I definitely kept the recent color temperature work from the openHAB side from @andrewfg. Let me know if I need to change any of the copyright/notices to better reference credit from the SmartHomeJ fork.

J-N-K commented on bundles/org.openhab.binding.amazonechocontrol/NOTICE:
```diff
@@ -11,3 +11,5 @@ https://www.eclipse.org/legal/epl-2.0/.
 
 https://github.com/openhab/openhab-addons
+
+Parts of this code have been forked from https://github.com/smarthomej/addons
```
Fine with me.

J-N-K commented on bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/ServletUri.java:
```diff
@@ -0,0 +1,52 @@
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault({})
```
Note to the reviewer: This is necessary because the eclipse compiler does not handle null annotations on records properly.

jimtng commented:
Amazing work, THANK YOU!

lsiepel commented:
This PR seems to have many conflicts, can you fix them @ccutrer ?

ccutrer commented:
> This PR seems to have many conflicts, can you fix them @ccutrer ?

Fixed. It was all the files removed in this PR that had the copyright updated in main. I rebased (and updated the copyright header in the new files).

lsiepel reviewed (approved):
Thanks @ccutrer for integrating all these changes mainly created by @J-N-K.
LGTM.

lsiepel commented:
> I think you need to also backport [smarthomej/addons@b337ff2](https://github.com/smarthomej/addons/commit/b337ff21f0c1d57e6e157da8565e32ae04fe2966) to make it work.

@ccutrer could i ask you to create a follow up PR ?

ccutrer commented:
> > I think you need to also backport [smarthomej/addons@b337ff2](https://github.com/smarthomej/addons/commit/b337ff21f0c1d57e6e157da8565e32ae04fe2966) to make it work.
> 
> @ccutrer could i ask you to create a follow up PR ?

Already on it ;). Just installed into my local openHAB to test, and committing now...
